srtcore/core.cpp

```diff
@@ -213,6 +216,31 @@ struct SrtOptionAction
 }
 srt_options_action;
 
+#if HAVE_CXX11
+
+CUDTUnited* CUDT::uglobal()
+{
+    static CUDTUnited instance;
+    return &instance;
+}
+
+#else // !HAVE_CXX11
+
+static pthread_once_t s_UDTUnitedOnce = PTHREAD_ONCE_INIT;
+static CUDTUnited*    s_UDTUnited     = NULL;
+
+static void createInstance()
+{
+    s_UDTUnited = new CUDTUnited();
+}
```

Comment:
How would the `CUDTUnited` object be destructed?

Reply:
I don't know how to delete the singleton automatically in a portable way.

`CUDT::cleanup` is a good entry to trigger delete, then another global lock is needed to protect CUDT::startup() and CUDT::cleanup() called by multiple threads.

What's your suggestion?

Reply:
How about this:
```
static CUDTUnited *getInstance()
{
    static CUDTUnited instance;
    return &instance;
}

static void createInstance() {
    getInstance();
}

CUDTUnited* srt::CUDT::uglobal()
{
    pthread_once(&s_UDTUnitedOnce, createInstance);
    return getInstance();
}
```

Reply:
Or just do cast and remove the createInstance()